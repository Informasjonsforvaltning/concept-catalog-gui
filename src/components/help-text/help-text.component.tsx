import React, { useState } from 'react';
import cx from 'classnames';
import { localization } from '../../lib/localization';
import { convertToSanitizedHtml } from '../../lib/markdown-converter';

import SC from './styled';

interface Props {
  title: string;
  helpTextAbstract?: string;
  helpTextDescription?: string;
  showRequired?: boolean;
}
/* eslint-disable react/no-danger */
export const HelpText = ({
  title,
  helpTextAbstract,
  helpTextDescription,
  showRequired = false
}: Props): JSX.Element => {
  const [collapse, setCollapse] = useState(false);

  const collapseClass = cx('fa', 'mr-2', {
    'fa-angle-double-down': !collapse,
    'fa-angle-double-up': collapse
  });

  const toggle = (): void => {
    setCollapse(!collapse);
  };

  return (
    <SC.HelpText>
      <div>
        <SC.Title className='help-text mb-0'>{title}</SC.Title>
        {showRequired && <SC.Required>{localization.required}</SC.Required>}
      </div>

      <div>
        {helpTextAbstract && (
          <p
            dangerouslySetInnerHTML={{
              __html: convertToSanitizedHtml(helpTextAbstract)
            }}
          />
        )}
        {helpTextDescription && (
          <SC.ShowMoreButton type='button' onClick={toggle}>
            <i className={collapseClass} />
            {collapse
              ? localization.lessRecomendations
              : localization.moreRecomendations}
          </SC.ShowMoreButton>
        )}
      </div>

      {helpTextDescription && (
        <SC.CollapseDescription isOpen={collapse}>
          <p
            dangerouslySetInnerHTML={{
              __html: convertToSanitizedHtml(helpTextDescription)
            }}
          />
        </SC.CollapseDescription>
      )}
    </SC.HelpText>
  );
};
/* eslint-enable react/no-danger */
