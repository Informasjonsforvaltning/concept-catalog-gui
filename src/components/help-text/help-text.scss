@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-reg-helptext {
  background-color: $color-primary-lighter;
  border-radius: 5px;
  font-size: 1.4rem;
  strong {
    font-size: 1.8rem;
  }
  .fdk-reg-helptext-more {
    text-decoration: underline;
    white-space: nowrap;
    color: $color-link;
  }
}

.help-text p {
  margin-bottom: 0;
}

.help-text {
  a {
    color: $color-link;
  }

  a:after {
    content: ' ';
    background-image: url('../../assets/img/icon-external-link-xs.svg');
    position: relative;
    width: 1.4em;
    height: 1.4em;
    top: +0.3em;
    display: inline-flex;
  }
}
