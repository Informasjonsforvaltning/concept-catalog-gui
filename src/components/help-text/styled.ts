import styled from 'styled-components';

import { Colour, theme } from '@fellesdatakatalog/theme';

const HelpText = styled.div`
  background-color: ${theme.colour(Colour.CYAN, 'C20')};
  border-radius: 5px;
  font-size: ${theme.fontSize('FS14')};
  margin-bottom: ${theme.spacing('S12')};
  padding: ${theme.spacing('S10')};
`;

const Title = styled.h3`
  font-size: ${theme.fontSize('FS20')};
`;

const Required = styled.div`
  background-color: ${theme.colour(Colour.GREEN, 'G60')};
  border-radius: 20px;
  color: ${theme.colour(Colour.NEUTRAL, 'N0')};
  margin-left: ${theme.spacing('S6')};
  padding: 0 ${theme.spacing('S6')};
`;

const ShowMoreButton = styled.button`
  align-self: flex-start;
  background-color: transparent;
  border: none;
  color: ${theme.colour(Colour.GREEN, 'G60')};
  margin-left: ${theme.spacing('S6')};
  text-decoration: underline;
  white-space: nowrap;
`;

export default { HelpText, Title, Required, ShowMoreButton };
