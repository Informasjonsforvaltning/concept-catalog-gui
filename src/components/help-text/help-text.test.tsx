import React from 'react';
import { shallow } from 'enzyme';
import { HelpText } from './help-text.component';

test('should render HelpText', (): void => {
  const defaultProps = {
    title: 'Title',
    helpTextAbstract: 'Abstract',
    helpTextDescription: 'Description',
    required: false
  };
  const wrapper = shallow(<HelpText {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
