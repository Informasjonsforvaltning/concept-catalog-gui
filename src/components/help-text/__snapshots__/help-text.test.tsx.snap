// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HelpText 1`] = `
<div
  className="fdk-reg-helptext mb-3 p-3"
>
  <div
    className="d-flex align-items-center"
  >
    <h3>
      Title
    </h3>
  </div>
  <div
    className="d-md-flex"
  >
    <p
      className="m-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Abstract</p>",
        }
      }
    />
    <button
      className="fdk-btn-no-border text-left p-0 ml-1 fdk-reg-helptext-more align-self-start"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fdk-fa-left fa-angle-double-down"
      />
      Hjelp
    </button>
  </div>
  <Collapse
    appear={false}
    className="mt-3"
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Description</p>",
        }
      }
    />
  </Collapse>
</div>
`;
